File: answers/leetcode/Sqrt(x).cpp
// binary search

class Solution {
public:
    int sqrt(int x) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
		if (x == 0)
		{
			return x;
		}
		int left = 1, right = x;
		while (left <= right)
		{
			int mid = left + (right - left) / 2;
			int div = x / mid;
			if (mid < div)
			{
				left = mid + 1;
			}
			else if (mid > div)
			{
				right = mid - 1;
			} else
			{
				return mid;
			}
		}
		return right;
    }
};

File: answers/leetcode/Sqrt(x).py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """

        left = 0
        right = x
        while left <= right:
            middle = (left + right) / 2
            sqrt = middle * middle
            if x == sqrt:
                return middle
            elif x < sqrt:
                right = middle - 1
            else:
                left = middle + 1
        return (left + right) / 2
